packages/core/src/Stores/common-store.js

```diff
@@ -62,6 +63,13 @@ export default class CommonStore extends BaseStore {
         }
     }
 
+    @action.bound
+    setLanguageChanging(is_language_changing) {
+        runInAction(() => {
+            this.is_language_changing = is_language_changing;
+        });
+    }
+
```

Comment:
immediate changing the is_language_changing status

Reply:
runInAction is useful in async functions but here it's unfortunately redundant. the following will give the same result 🙂
```suggestion
setIsLanguageChanging(is_language_changing) {
            this.is_language_changing = is_language_changing;
    }
```

Reply:
Thanks for this great idea. I only moved timer_id outside this function, otherwise  we will create new variable timer_id with every function call, and will not be able to clear previous timer